packages-exp/auth-exp/src/mfa/mfa_user.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as externs from '@firebase/auth-types-exp';
+
+import { User } from '../model/user';
+import { MultiFactorSessionType, MultiFactorSession } from './mfa_session';
+import { MultiFactorAssertion } from './assertions';
+import { withdrawMfa } from '../api/account_management/mfa';
+import { AuthErrorCode } from '../core/errors';
+
+export class MultiFactorUser implements externs.MultiFactorUser {
+  // TODO(avolkovi): Set these correctly from getAccountInfo on reload
+  enrolledFactors: externs.MultiFactorInfo[] = [];
+
+  constructor(readonly user: User) {}
+
+  async getSession(): Promise<MultiFactorSession> {
+    return new MultiFactorSession(
+      MultiFactorSessionType.ENROLL,
+      await this.user.getIdToken()
+    );
+  }
+
+  async enroll(
+    assertionExtern: externs.MultiFactorAssertion,
+    displayName?: string | null
+  ): Promise<void> {
+    const assertion = assertionExtern as MultiFactorAssertion;
+    const session = await this.getSession();
+    const idTokenResponse = await assertion._process(session, displayName);
```

Comment:
Double check that displayName goes through properly. "Undefined" means it should be missing from the update, null is meant to clear it out

Reply:
it's undefined | string | null all the way through to the API, will verify for sure once I get the demo up and running